rcl_logging_log4cxx/include/rcl_logging_log4cxx/logging_interface.h

```diff
@@ -23,17 +23,54 @@ extern "C" {
 
 typedef int rcl_logging_ret_t;
 
+/// Initialize log4cxx logging library.
+/*
+ * \param[in] config_file The location of a config file that the external
+ *   logging library should use to configure itself.
+ *   If no config file is provided this will be set to an empty string.
+ *   Must be a NULL terminated c string.
+ * \param[in] allocator The allocator to use for memory allocation. This is
+ *   an rcutils_allocator_t rather than an rcl_allocator_t to ensure that the
+ *   rcl_logging_* packages don't have a circular dependency back to rcl.
+ * \return RCL_LOGGING_RET_OK if initialized successfully, or
+ * \return RCL_LOGGING_RET_ERROR if an unspecified error occurs.
+ */
 RCL_LOGGING_PUBLIC
 rcl_logging_ret_t rcl_logging_external_initialize(
   const char * config_file,
   rcutils_allocator_t allocator);
 
+/// Free the resources allocated for the log4cxx external logging system.
+/**
+ * This puts the system into a state equivalent to being uninitialized.
+ *
+ * \return always RCL_LOGGING_RET_OK.
+ */
 RCL_LOGGING_PUBLIC
 rcl_logging_ret_t rcl_logging_external_shutdown();
 
+/// Log a message.
+/**
+ * \param[in] severity The severity level of the message being logged.
+ * \param[in] name The name of the logger, must either be a null terminated
+ *   c string or NULL.
+ *   If NULL or empty the root logger will be used.
+ * \param[in] msg The message to be logged. Must be a null terminated c string.
+ */
 RCL_LOGGING_PUBLIC
 void rcl_logging_external_log(int severity, const char * name, const char * msg);
 
+/// Set the severity level for a logger.
+/**
+ * This function sets the severity level for the specified logger.
+ * If the name provided is an empty string or NULL it will change the level of
+ * the root logger.
+ *
+ * \param[in] name The name of the logger.
+ *   Must be a NULL terminated c string or NULL.
+ * \param[in] level The severity level to be used for the specified logger.
```

Comment:
Where are these severity levels defined? Maybe you can link to them or tag them?